test/binder/binder_test.cpp

```diff
@@ -0,0 +1,200 @@
+  tupleExpr = (expression::TupleValueExpression*)(*selectStmt->select_list)[1];
+  EXPECT_EQ(tupleExpr->BoundObjectId,
+            make_tuple(db_oid, tableB_oid, 1));  // B.b2
+
+  // Check join condition
+  LOG_INFO("Checking join condition");
+  tupleExpr = (expression::TupleValueExpression*)
+                  selectStmt->from_table->join->condition->GetChild(0);
+  EXPECT_EQ(tupleExpr->BoundObjectId,
+            make_tuple(db_oid, tableA_oid, 0));  // a.a1
+  tupleExpr = (expression::TupleValueExpression*)
+                  selectStmt->from_table->join->condition->GetChild(1);
+  EXPECT_EQ(tupleExpr->BoundObjectId,
+            make_tuple(db_oid, tableB_oid, 0));  // b.b1
+
+  // Check Where clause
+  LOG_INFO("Checking where clause");
+  tupleExpr =
+      (expression::TupleValueExpression*)selectStmt->where_clause->GetChild(0);
+  EXPECT_EQ(tupleExpr->BoundObjectId, make_tuple(db_oid, tableA_oid, 0));  // a1
+
+  // Check Group By and Having
+  LOG_INFO("Checking group by");
+  tupleExpr =
+      (expression::TupleValueExpression*)selectStmt->group_by->columns->at(0);
+  EXPECT_EQ(tupleExpr->BoundObjectId,
+            make_tuple(db_oid, tableA_oid, 0));  // A.a1
+  tupleExpr =
+      (expression::TupleValueExpression*)selectStmt->group_by->columns->at(1);
+  EXPECT_EQ(tupleExpr->BoundObjectId,
+            make_tuple(db_oid, tableB_oid, 1));  // B.b2
+  tupleExpr =
+      (expression::TupleValueExpression*)selectStmt->group_by->having->GetChild(
+          0);
+  EXPECT_EQ(tupleExpr->BoundObjectId, make_tuple(db_oid, tableA_oid, 0));  // a1
+
+  // Check Order By
+  LOG_INFO("Checking order by");
+  tupleExpr = (expression::TupleValueExpression*)selectStmt->order->expr;
+  EXPECT_EQ(tupleExpr->BoundObjectId, make_tuple(db_oid, tableA_oid, 0));  // a1
+
+  // Check alias ambiguous
+  LOG_INFO("Checking duplicate alias and table name.");
+  binder.reset(new binder::BindNodeVisitor());
+  selectSQL = "SELECT * FROM A, B as A";
+  parse_tree = parser.BuildParseTree(selectSQL);
+  selectStmt = (parser::SelectStatement*)(parse_tree->GetStatements().at(0));
+  try {
+    binder->BindNameToNode(selectStmt);
+    EXPECT_TRUE(false);
+  } catch (Exception& e) {
+    LOG_INFO("Correct! Exception(%s) catched", e.what());
+  }
+
+  // Test alias and select_list
+  LOG_INFO("Checking select_list and table alias binding");
+  binder.reset(new binder::BindNodeVisitor());
+  selectSQL = "SELECT AA.a1, b2 FROM A as AA, B WHERE AA.a1 = B.b1";
+  parse_tree = parser.BuildParseTree(selectSQL);
+  selectStmt = (parser::SelectStatement*)(parse_tree->GetStatements().at(0));
+  binder->BindNameToNode(selectStmt);
+  tupleExpr =
+      (expression::TupleValueExpression*)(selectStmt->select_list->at(0));
+  EXPECT_EQ(tupleExpr->BoundObjectId, make_tuple(db_oid, tableA_oid, 0));
+  tupleExpr =
+      (expression::TupleValueExpression*)(selectStmt->select_list->at(1));
+  EXPECT_EQ(tupleExpr->BoundObjectId, make_tuple(db_oid, tableB_oid, 1));
+
```

Comment:
Can you add a test case for something like "select * from A, A as AA where A.a1 = AA.a2"?